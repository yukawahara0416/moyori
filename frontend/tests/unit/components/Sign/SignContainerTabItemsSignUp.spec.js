import { mount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import Component from '@/components/Sign/SignContainerTabItemsSignUp.vue'

const localVue = createLocalVue()
localVue.use(Vuex)

let wrapper
let getters
let actions
let store

beforeEach(() => {
  getters = {
    signUpFormData: () => ({
      name: 'tester',
      email: 'tester@example.com',
      password: 'password',
      password_confirmation: 'password'
    }),
    currentUser: () => ({ data: { id: 1 } })
  }

  actions = {
    signUp: jest.fn()
  }

  store = new Vuex.Store({
    getters,
    actions
  })

  wrapper = mount(Component, {
    localVue,
    store,
    stubs: ['policy']
  })
})

afterEach(() => {
  wrapper.destroy()
})

describe('getters', () => {
  it('signUpFormData', () => {
    expect(wrapper.vm.signUpFormData).toEqual(getters.signUpFormData())
  })
  it('currentUser', () => {
    expect(wrapper.vm.currentUser).toEqual(getters.currentUser())
  })
})

describe('v-on', () => {
  it('signUp', () => {
    const event = jest.fn()
    wrapper.setMethods({ signUp: event })
    wrapper.find('.v-btn').trigger('click')
    expect(event).toHaveBeenCalledTimes(1)
  })
})

describe('actions', () => {
  it('signUp', () => {
    wrapper.vm.signUp()
    expect(actions.signUp).toHaveBeenCalled()
    // expect(actions.signUp).toHaveBeenCalledWith(
    //   expect.any(Object),
    //   getters.signUpFormData,
    //   undefined
    // )
  })
})

describe('template', () => {
  it('snapshot', () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
