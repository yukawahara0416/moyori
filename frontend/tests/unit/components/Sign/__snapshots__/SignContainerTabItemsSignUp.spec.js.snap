// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template snapshot 1`] = `
<div
  class="container container--fluid"
>
  <div
    class="row align-center justify-center"
    style="flex-direction: column;"
  >
    <div
      class="col"
    >
      <div
        class="mx-5 v-card v-sheet theme--light"
      >
        <header
          class="white--text v-sheet v-sheet--tile theme--light v-toolbar v-toolbar--dense v-toolbar--flat primary"
          style="height: 48px;"
        >
          <div
            class="v-toolbar__content"
            style="height: 48px;"
          >
            <div
              class="v-toolbar__title"
            >
              SNSアカウントで登録
            </div>
          </div>
        </header>
         
        <div
          class="v-card__actions"
        >
          <div
            class="row align-center justify-center"
            style="flex-direction: column;"
          >
            <button
              class="my-3 white--text v-btn v-btn--contained theme--light v-size--large"
              style="background-color: rgb(0, 172, 238); border-color: #00acee;"
              type="submit"
            >
              <span
                class="v-btn__content"
              >
                <i
                  aria-hidden="true"
                  class="v-icon notranslate v-icon--left mdi mdi-twitter theme--light white--text"
                />
                
              Twitterアカウントで新規登録
            
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="col"
    >
      <div
        class="mx-5 v-card v-sheet theme--light"
      >
        <header
          class="white--text v-sheet v-sheet--tile theme--light v-toolbar v-toolbar--dense v-toolbar--flat primary"
          style="height: 48px;"
        >
          <div
            class="v-toolbar__content"
            style="height: 48px;"
          >
            <div
              class="v-toolbar__title"
            >
              メールアドレスで登録
            </div>
          </div>
        </header>
         
        <div
          class="v-card__text"
        >
          <form
            class="v-form"
            novalidate="novalidate"
          >
            <div
              class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field"
            >
              <div
                class="v-input__prepend-outer"
              >
                <div
                  class="v-input__icon v-input__icon--prepend"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate mdi mdi-account-circle-outline theme--light"
                  />
                </div>
              </div>
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label v-label--active theme--light"
                      for="input-127"
                      style="right: 0px; position: absolute;"
                    >
                      名前
                    </label>
                    <input
                      id="input-127"
                      name="signin"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field"
            >
              <div
                class="v-input__prepend-outer"
              >
                <div
                  class="v-input__icon v-input__icon--prepend"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate mdi mdi-email-outline theme--light"
                  />
                </div>
              </div>
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label v-label--active theme--light"
                      for="input-131"
                      style="right: 0px; position: absolute;"
                    >
                      メールアドレス
                    </label>
                    <input
                      id="input-131"
                      name="signin"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field"
            >
              <div
                class="v-input__prepend-outer"
              >
                <div
                  class="v-input__icon v-input__icon--prepend"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate mdi mdi-lock-outline theme--light"
                  />
                </div>
              </div>
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label v-label--active theme--light"
                      for="password"
                      style="right: 0px; position: absolute;"
                    >
                      パスワード
                    </label>
                    <input
                      id="password"
                      name="password"
                      type="password"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field"
            >
              <div
                class="v-input__prepend-outer"
              >
                <div
                  class="v-input__icon v-input__icon--prepend"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate mdi mdi-lock-outline theme--light"
                  />
                </div>
              </div>
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label v-label--active theme--light"
                      for="password_confirmation"
                      style="right: 0px; position: absolute;"
                    >
                      パスワード（確認用）
                    </label>
                    <input
                      id="password_confirmation"
                      name="password_confirmation"
                      type="password"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                </div>
              </div>
            </div>
          </form>
        </div>
         
        <div
          class="v-card__actions"
        >
          <div
            class="spacer"
          />
           
          <button
            class="mb-3 px-10 v-btn v-btn--contained theme--light v-size--large primary"
            type="submit"
          >
            <span
              class="v-btn__content"
            >
              
            登録
          
            </span>
          </button>
           
          <div
            class="spacer"
          />
        </div>
      </div>
    </div>
     
    <policy-stub />
  </div>
</div>
`;
